### DISCUSSION TITLE
Config ignored when creating a "higher level" client from a "lower level" client

### DISCUSSION THREAD

**Discussion by gspetrou** [aws/aws-sdk-js-v3#5472] [Q&A]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

Consider the following code, imagining that we're running from a Lambda in `us-west-2`.

```
const lowerLevel = new S3Client({region: 'us-west-1'})
const higherLevel = new S3(lowerLevel)

// error: PermanentRedirect: The bucket you are attempting to access must be addressed using the specified endpoint.
// Please send all future requests to this endpoint.
await s3.getObject({Bucket: 'my-bucket', Key: 'my-key'})
```

When creating a "v2 style" (or "higher level") client by passing in a "v3-style" (or "lower level") client, the config parameters of the lower-level client are ignored by the higher level client. In the example above, the GetObject API call is made using the `us-west-2` S3 endpoint where the Lambda is running from, rather than `us-west-1` where the bucket exists. Resulting in a "PermanentRedirect" error.

### SDK version number

@aws-sdk/client-s3@^3.427.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

Node 18 Lambda

### Reproduction Steps

```
import {S3, S3Client} from '@aws-sdk/client-s3'

const lowerLevel = new S3Client({region: 'us-west-1'})
const higherLevel = new S3(lowerLevel)

// Fails
const res = await s3.getObject({Bucket: 'my-bucket', Key: 'my-key'})
```

### Observed Behavior

Error thrown:

```
error: PermanentRedirect: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.
    at xde (/node_modules/@smithy/smithy-client/dist-cjs/default-error-handler.js:8:22)
    at throwDefaultError (/node_modules/@smithy/smithy-client/dist-cjs/default-error-handler.js:18:21)
    at NNe (/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:4330:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at <anonymous> (/node_modules/@smithy/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24)
    at <anonymous> (/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/middleware-signing/dist-cjs/awsAuthMiddleware.js:14:20)
    at <anonymous> (/node_modules/@smithy/middleware-retry/dist-cjs/retryMiddleware.js:27:46)
    at <anonymous> (/node_modules/@aws-sdk/middleware-flexible-checksums/dist-cjs/flexibleChecksumsMiddleware.js:57:20)
    at <anonymous> (/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/region-redirect-endpoint-middleware.js:14:24)
    at <anonymous> (/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/region-redirect-middleware.js:9:20) {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 301,
    requestId: '4G8HRMRE0PV5HNPH',
    extendedRequestId: '<omitted>',
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  Code: 'PermanentRedirect',
  Endpoint: '<omitted>.s3-us-west-1.amazonaws.com',
  Bucket: '<omitted>',
  RequestId: '4G8HRMRE0PV5HNPH',
  HostId: '<omitted>'
}
```

Notice that the `endpoint` states the correct S3 endpoint, however when inspecting the entire `S3` client object I noticed the following:

```
          freeSockets: [Object: null prototype] {
            '<bucketName>.s3.us-west-2.amazonaws.com:443:::::::::::::::::::::': [
              <ref *1> TLSSocket {
                ...
```

### Expected Behavior

The GetObject call goes to `us-west-1` and succeeds

### Additional Information/Context

Feel free to reach out internally as well, I'm an AWS employee (`gspetrou`)

**Comment by RanVaknin** [comment]
Hi @gspetrou ,

Thanks for opening the issue. I think there might be some confusion regarding the differences between S3 and S3client. 
The intention is for you to use one or the other, and not create one from the other.

Both S3 and S3client take the same config type as input, and not each other as input. 

So either use the modular style SDK client

```js
import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3";

const s3Client = new S3Client({region: 'us-west-1'});

try {
    const response = await s3Client.send(new GetObjectCommand({ Bucket: 'my-bucket', Key: 'my-key'}));
    console.log(response);
} catch (err){
    console.log(err);
}
```

Or use the "v2-style" client:

```js
import { S3 } from "@aws-sdk/client-s3";

const s3 = new S3({ region: 'us-west-1' });

try {
    const response = await s3.getObject({ Bucket: 'my-bucket', Key: 'my-key' }).promise();
    console.log(response);
} catch (err) {
    console.error(err);
}
```

I hope this clears up any confusion. Let us know if you have further questions or issues.

Thanks,
Ran~
